### DISCUSSION TITLE
Add server-based Jamulus Mixer

### DISCUSSION THREAD

**Discussion by denoveri** [jamulussoftware/jamulus#1632] [Ideas]
Presently, each Jamulus client has a mixer for local sound balancing.  It would be beneficial to have a mixer for the session on the server and allow a client to control the server-based faders.  This way everyone would get the same balanced mix.  Since a lot of users configure for two mono devices (e.g. microphone and guitar), each side of a stereo channel should have its own fader.

Client-based local mixers could still exist for all the clients.

It would be ideal to permit control of the server faders via a MIDI controller keyboard.
Thus, a keyboardist could do dual duty as player and sound person.

ALREADY BEING DONE IN ANOTHER INITIATIVE -- NEEDS TO BE DELETED

**Comment by dcorson-ticino-com** [comment]
Hi Denoveri,
There are at present two active projects (to my knowledge) to add such a "master mix" to Jamulus using different guiding concepts.
You can look in the pull requests to see how they are advancing and try them out.

Note that technically the Jamulus clients do not do any mixing.  The clients send the information about their fader positions to the server and the server does all the mixing, separately for each client.
There is also already MIDI support in Jamulus, please look in the documention at Jamulus.io, I do not know the details.

**Reply by cwerling** [reply]
Shameless plug: https://github.com/jamulussoftware/jamulus/pull/1381

**Comment by denoveri** [comment]
Being a newbie, I didn't know how to search well for initiatives already in progress doing something similar.  So now I think this conversation should be deleted.  How do we do this?

**Reply by dcorson-ticino-com** [reply]
No harm done.  It is always difficult to know what to search for.

**Reply by hoffie** [reply]
I think it's ok and maybe even beneficial to keep this discussion. If someone else searches for such a feature, they might rather use key words from this discussion, so it might help. :)
